### DISCUSSION TITLE
Batch Editor randomize ability

### DISCUSSION THREAD

**Discussion by Phyore** [kwsch/PKHeX#3586] [Q&A]
Attemping to use the Batch Editor to set a random ability onto each of my party pokemon. I am attempting to use 
`.Ability=$rand`
But when it runs it seems to have no affect on the party. Is there a better way to go about that?

**Comment by kwsch** [comment]
$rand is only used for 32bit values like PID.

Refer to the batch editor guide on how to randomize values within a specified range.

**Reply by Phyore** [reply]
> $rand is only used for 32bit values like PID.
> 
> Refer to the batch editor guide on how to randomize values within a specified range.

Super late response to this (My bad) I am trying the following but it does not seem to be setting it on the Pokemon in that box. I tried the below so it would select a random one between 1 and 267. It does however mention that is has modified a Pokemon.

```
=Box=31
.Ability=1,267
```

**Reply by kwsch** [reply]
The single PKM editor auto fixes things like wrong abilities. Run the batch edit, then export your save and check in game.

**Reply by Phyore** [reply]
![image](https://user-images.githubusercontent.com/2222368/197811521-83eba81a-35e0-42d7-8829-ed34ea698c80.png)
Checking in game after the save also shows it with the default ability in this case. That is after running the above command through the batch editor

**Reply by sora10pls** [reply]
Your syntax is wrong.
```
=Box=31
.Ability=$1,267
```

EDIT: Actually, running this specific command seems to have PKHeX automatically correct the Ability afterwards.

Running `.Ability=69` will correctly set all PKMs to have the Ability Rock Head, but running `.Ability=$x,y` will correct their Ability to be whatever their second standard Ability is.

**Reply by Phyore** [reply]
> Your syntax is wrong.
> 
> ```
> =Box=31
> .Ability=$1,267
> ```
> 
> EDIT: Actually, running this specific command seems to have PKHeX automatically correct the Ability afterwards.
> 
> Running `.Ability=69` will correctly set all PKMs to have the Ability Rock Head, but running `.Ability=$x,y` will correct their Ability to be whatever their second standard Ability is.

Yeah, from what I see `$1` will set it to first ability and `$2` the second (From a legit pokemon) but trying to get it to use a random number between 1 and 267 doesn't seem to work. As you mentioned `.Ability=69` will function correctly.

It mostly feels like it might not be using that "pick a number between" correctly (Or at least I am doing something wrong)

**Reply by kwsch** [reply]
Revised logic in latest commit.
```
.Ability=$1,267
```
^ now works to randomize between 1 and 267.

**Reply by Phyore** [reply]
I will either work out how to build it or download the release when ready! Thank you!

**Reply by Phyore** [reply]
Works great, thanks!
